File: frontend/src/types/inventory.ts
// Backend-aligned inventory types.
export type Operation = 'STOCK_IN' | 'STOCK_OUT' | 'ADJUSTMENT';

export interface Transaction {
  id: string;
  product_id: string;
  product_name?: string;
  operation: Operation;
  quantity: number;
  unit: string;
  source: 'voice' | 'manual' | 'system';
  reason?: string;
  price_total?: number;
  created_at: string;
}

export interface StockBalance {
  product_id: string;
  product_name?: string;
  quantity: number;
  unit: string;
  reorder_threshold?: number;
  status?: string;
}

export interface DashboardProduct {
  id: string;
  name: string;
  quantity: number;
  unit: string;
  reorder_threshold: number;
  category?: string;
  status: string;
}

export interface DashboardSummary {
  today_stock_in: number;
  today_stock_out: number;
  low_stock_count: number;
  total_products: number;
  products: DashboardProduct[];
}

File: backend/app/schemas/inventory.py
from datetime import datetime
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel
from app.core.constants import OperationType

class TransactionCreate(BaseModel):
    product_id: str
    operation: OperationType
    quantity: Decimal
    unit: str
    reason: Optional[str] = None
    price_total: Optional[Decimal] = None
    client_request_id: Optional[str] = None

class TransactionResponse(BaseModel):
    id: str
    product_name: str
    operation: str
    quantity: float
    unit: str
    new_balance: float
    created_at: datetime

class StockBalanceResponse(BaseModel):
    product_id: str
    product_name: str
    quantity: float
    unit: str
    reorder_threshold: float
    status: str

class DashboardSummary(BaseModel):
    today_in: int
    today_out: int
    low_stock_count: int
    total_products: int
    products: List[StockBalanceResponse]
